content/post/deduplicating_pga_with_apollo.md

```diff
@@ -0,0 +1,398 @@
+Average ratio of distinct file names per file in communities depending on 
+the minimum number of files, at 80% (left)and 95% (right) thresholds.  
+{{% /caption %}}
+
+### Embedded dependencies
+
+{{% caption src="/post/deduplicating_pga_with_apollo/text_parser_go.png" %}}
+Graph of the connected component with `text_parser.go` files described below, colored by community.
+Buckets are colored in black, their size depends of the number of edges.
+{{% /caption %}}
+
+| files count | buckets count | edges count | distinct filenames count | projects count | communities count | threshold |
+|-----------|---------------|-------------|--------------------------|----------------|-------------------|-----------|
+| 422 | 371 | 1266 | 1 | 327 | 2 |  95% | 
+
+This is just one of the 79,000 connected components with a single distinct Go file name
+even without FuzzyWuzzy reduction. Quick investigation revealed that they probably
+come from vendoring different versions of [protobuf](https://github.com/golang/protobuf).
+
+
+### Versioning
+
+{{% caption src="/post/deduplicating_pga_with_apollo/jquery.png" %}}
+Graph of the connected component with jQuery files described below, colored by community.
+Buckets are colored in black, their size depends of the number of edges.
+{{% /caption %}}
+
+| files count | buckets count | edges count | distinct filenames count | projects count | communities count | threshold |
+|-----------|---------------|-------------|--------------------------|----------------|-------------------|-----------|
+| 1532 | 644 | 4596 | 238  | 796 | 37|  95% | 
+
+
+Not only did we notice that CCs tended to group files with matching names,
+we also noticed that there were multiple CCs with variations of the same file name.
+For example, we found 14 CCs with different blends of `jquery.js`, e.g. `jquery-1.7.1.min.js`.
+Those CCs typically had a small number of communities for their size, however,
+they clustered by file name to very limited extent.
+
+### Are file names so important?
+
+{{% caption src="/post/deduplicating_pga_with_apollo/filename_1.png" %}}
+Graph of the connected component with Ruby files described below, colored by community.
+Edges are colored depending on the vertices they link to.
+{{% /caption %}}
+{{% caption src="/post/deduplicating_pga_with_apollo/filename_2.png" %}}
+Graph of the connected component with Ruby files described below,
+colored by file name (see the legend). 
+Files with rare (less than 1%) names are colored with grey.
+{{% /caption %}}
+
+| files count | edges count | distinct filenames count | projects count | communities count | threshold |
+|-------------|-------------|--------------------------|----------------|-------------------|-----------|
+| 2344        | 869,611     | 584                      | 1058           | 3                 |  80%      | 
+
+This CC does not contain artificial vertices - buckets - because as we wrote in section "Landing"
```

Comment:
Make `"Landing"` a link to the section, I think you can point to it by using `[Landing](#landing)`